capstone/src/main/webapp/script.js

```diff
@@ -1 +1,261 @@
-//some js code here
+async function postSurveyResponses(){
+    let response = await fetch('/data', {method: "POST"});
+}
+
+function loadCharts(){
+    let stats = new google.visualization.DataTable();
+    
+    stats.addColumn('string', 'Sentiment');
+    stats.addColumn('number', 'Percentage');
+    stats.addRows([
+        ['Positive', 0.7],
+        ['Neutral', 0.1],
+        ['Negative', 0.2],
+    ]);
+
+    // Instantiate and draw the chart.
+    const chart = new google.visualization.PieChart(document.getElementById('sentiment-pie-chart'));
+    chart.draw(stats, null);
+
+    loadReponseChart();
+}
+
+function loadReponseChart(){
+    let stats = new google.visualization.DataTable();
+    
+    stats.addColumn('string', 'Period');
+    stats.addColumn('number', 'Number of Responses');
+
+    stats.addRows([
+        ['Daily', 4],
+        ['Weekly', 15]
+    ]);
+
+    // Instantiate and draw the chart.
+    const chart = new google.visualization.BarChart(document.getElementById('response-bar-chart'));
+    chart.draw(stats, null);
+}
+
+/* Statistics Page Methods */
+
+async function getSurveyResponses(){
+    let response = await fetch('/load-data');
+    let list = await response.json(); //list of entities from datastore
+    let scores = new Array();
+    let sentimentCount = [0,0,0,0,0]; // Very positive, positive, neutral, negative, and very negative
+    const refGender = ['M','F','U'];
+    let genderCount = [0,0,0]; // Male, female, and unknown
+    const refAge = ['18-24','25-34','35-44','45-54','55-64','65+','Unknown'];
+    let ageCount = [0,0,0,0,0,0,0]; // 18-24, 25-34, 35-44, 45-54, 55-64, 65+, or Unknown
```

Comment:
I believe the implementation is the same since I am using the "ref" arrays to get constant time access to each element so I can update their counters. Also, using arrays in this case make it easier to work with, since I have to pass them to separate functions that build the charts through a data table object. It's possible to use a map instead, but it may take a lot more time to configure the charts to read through iterable map objects the way I have it set up now.

Reply:
I don't think the benefit here is efficiency, its readability. You are essentially using a map without writing it as one. Using a map might make what you are doing cleaner and easier to read.